proto/stream.proto

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2017-2023 PixelsDB.
+ *
+ * This file is part of Pixels.
+ *
+ * Pixels is free software: you can redistribute it and/or modify
+ * it under the terms of the Affero GNU General Public License as
+ * published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version.
+ *
+ * Pixels is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * Affero GNU General Public License for more details.
+ *
+ * You should have received a copy of the Affero GNU General Public
+ * License along with Pixels.  If not, see
+ * <https://www.gnu.org/licenses/>.
+ */
+
+// grpc protocol of streaming over HTTP, for operator pipelining
+
+syntax = "proto2";
+
+package stream.proto;
+
+option java_multiple_files = false;
+option java_package = "io.pixelsdb.pixels.turbo";
+option java_outer_classname = "StreamProto";
+
+/**
+ * DESIGN: In streaming mode, we cannot determine all these footer fields until the end of the stream. So:
+ * In streaming mode, the whole message flow is like:
+ * ------------------------------------------------------------------
+ * | magic | streamHeaderLen | streamHeader | rowGroup | rowGroup | ... |
+ * ------------------------------------------------------------------
+ * where each rowGroup is sent like:
+ * -----------------------------------------------------------------------
+ * | rowGroupHeaderLen | rowGroupHeader | rowGroupDataLen | rowGroupData |
+ * -----------------------------------------------------------------------
+ * -----------------------------------------------------------------------
+ * | rowGroupDataLen | rowGroupData | rowGroupFooterLen | rowGroupFooter |
```

Comment:
How rowGroupHeader is defined? Why do we need a rowGroupFooter? Why not combine it with the rowGroupHeader?